File: tests/go/cases/case_031_archive_profile.go
package escape_tests

var retainedCase031 = []map[string]string{}

func Case031ArchiveProfile(input string) string {
    // Task case 031: archive profile payload for downstream business processing.
    raw := input
    if raw == "" {
        raw = "sample"
    }
    payload := map[string]string{
        "task": "archive_profile",
        "entity": "profile",
        "stage": "archive",
        "input": raw,
    }
    // ESCAPE: payload is promoted to package-level retained cache.
    retainedCase031 = append(retainedCase031, payload)
    return "ok"
}

File: tests/java/src/main/java/com/escape/tests/cases/Case031ArchiveProfile.java
package com.escape.tests.cases;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.function.Supplier;

/** Task case 031: archive profile payload for downstream business processing. */
public class Case031ArchiveProfile {
    private static final Map<String, Map<String, String>> RETAINED_CACHE = new HashMap<>();
    private static final List<Map<String, String>> RETAINED_AUDIT = new ArrayList<>();
    private static final List<Supplier<String>> RETAINED_HANDLERS = new ArrayList<>();

    public static String execute(String input) {
        // Task: archive profile records and prepare transport-ready payload.
        String taskName = "archive_profile";
        String raw = (input == null || input.isEmpty()) ? "sample" : input;
        Map<String, String> payload = new HashMap<>();
        payload.put("task", taskName);
        payload.put("entity", "profile");
        payload.put("stage", "archive");
        payload.put("input", raw);
        // ESCAPE: payload is promoted to class-level retained cache.
        RETAINED_CACHE.put("case_031", payload);
        return "ok";
    }
}
